File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.go
func topKFrequent(nums []int, k int) []int {
    count := make(map[int]int, len(nums))
    for _, n := range nums {
        count[n]++
    }
    
    inverse := make(map[int][]int, len(count))
    for k, v := range count {
        if s, ok := inverse[v]; ok {
            inverse[v] = append(s, k)
        } else {
            inverse[v] = []int{k}
        }
    }
    
    result := make([]int, 0, k)
    i := len(nums)
    for len(result) < k {
        if v, ok := inverse[i]; ok {
            result = append(result, v...)
        }
        i--
    }
    
    return result
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        for n in nums:
            count[n] = count.get(n, 0) + 1
        
        inverse = {}
        for key, val in count.items():
            inverse[val] = inverse.get(val, []) + [key]
        
        result = []
        i = len(nums)
        while len(result) < k:
            if inverse.get(i) is not None:
                result += inverse[i]
            i -= 1    
        
        return result